cli/src/utilities/generateNWUI.ts

```diff
@@ -0,0 +1,58 @@
+import { GluegunToolbox } from 'gluegun';
+import { CliResults } from '../types';
+import { nativeWindUIOptions } from '../constants';
+import { getPackageManagerRunnerX } from './getPackageManager';
+import { ONLY_ERRORS, runSystemCommand } from './systemCommand';
+import { spinner } from '@clack/prompts';
+
+export async function generateNWUI(cliResults: CliResults, toolbox: GluegunToolbox) {
+  const isNativeWindUISelected = cliResults.packages.some((p) => p.name === 'nativewindui');
+
+  if (!isNativeWindUISelected) {
+    return;
+  }
+
+  const s = spinner();
+
+  const runnerType = getPackageManagerRunnerX(toolbox, cliResults);
+
+  const nativeWindUIComponents =
+    cliResults.packages.find((p) => p.name === 'nativewindui').options.selectedComponents ?? [];
+
+  // we do this to account for older stored config e.g that has selectable text in it
+  const onlySupportedComponents = nativeWindUIComponents.filter((component) => nativeWindUIOptions.includes(component));
+
+  const finalComponents = Array.from(new Set([...onlySupportedComponents, 'text']));
+
+  s.start(`Adding nativewindui components...`);
+
+  const flags = cliResults.flags.noInstall
```

Comment:
this is the other fix in this PR (relative to #390 which this PR is a redo of): no extra space in the `nwui-cli` args if `--no-install` is not provided